_posts/2020-11-08-introduction-to-functions-in-R.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title:  "Introduction to programming functions in R!"
+date:   2020-11-08 12:54:04 -0700
+categories: R topics
+---
+In this post you will learn the basics of functions in R.  This will include:
+what a function is, how to set up a function, and how to call for a function
+from another R script.
+
+# What is a function?
+In programming, a function is a subroutine that performs some calculation with
+given variables.  These variables are given by a program and a result is
+returned.
+
+You can think of this type of function the same way you would think about a
+mathematical one.  
+The function is a shorthand way to make different variables go through the
+same process without needing to rewrite the formula every time.  
+
+For instance: if you have the function y=x^2, and are given a list of x
+variables (1:5).  You are doing the same function 5 times with different
+variables giving different outputs.  Here the function would be running in
+a loop of y=1^2, y=2^2, y=3^2, y=4^2, and y=5^2.  
+
+The difference between this mathematical function and a programming function
+that you make is that the programming function is located on its own R script
```

Comment:
So functions don't have to be in their own scripts, that's just one way to help keep things organized and allow them to be used in multiple other places, so I might adjust the wording a little here.